Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeSystemComponent.cpp

```diff
@@ -112,4 +124,54 @@ namespace AzToolsFramework
         return AzToolsFramework::IsInFocusSubTree(entityId, m_focusRoot);
     }
 
+    void FocusModeSystemComponent::OnEntityInfoUpdatedAddChildEnd(AZ::EntityId parentId, AZ::EntityId childId)
+    {
+        // If the parent's entityId is in the list, add the child.
+        if (auto iter = AZStd::find(m_focusedEntityIdList.begin(), m_focusedEntityIdList.end(), parentId);
+            iter != m_focusedEntityIdList.end())
+        {
+            m_focusedEntityIdList.push_back(childId);
+        }
+    }
+
+    void FocusModeSystemComponent::OnEntityInfoUpdatedRemoveChildEnd([[maybe_unused]] AZ::EntityId parentId, AZ::EntityId childId)
+    {
+        // If the removed entityId is in the list, remove it.
+        if (auto iter = AZStd::find(m_focusedEntityIdList.begin(), m_focusedEntityIdList.end(), childId);
+            iter != m_focusedEntityIdList.end())
+        {
+            m_focusedEntityIdList.erase(iter);
```

Comment:
suggestion: Right now as the ordering is not important, we might want to do a `swap` and `pop_back` with the last element to stop the expensive reshuffle happening if an entity is right in the middle of the list.